mw-webapp/src/dataAccessLogic/DayReportDAL.ts

```diff
@@ -23,8 +22,27 @@ export class DayReportDAL {
   public static async getDayReports(wayUuid: string): Promise<DayReport[]> {
     const dayReportsUuids = (await WayService.getWayDTO(wayUuid)).dayReportUuids;
 
-    const dayReports = await Promise.all(dayReportsUuids.map(async (dayReportUuid) => {
-      const dayReport = await DayReportDAL.getDayReport(dayReportUuid);
+    const dayReportsDTO = await DayReportService.getDayReportsDTO(dayReportsUuids);
+
+    const dayReports = await Promise.all(dayReportsDTO.map(async (dayReportDTO) => {
+      const {jobDoneUuids, planForNextPeriodUuids, commentUuids, problemForCurrentPeriodUuids} = dayReportDTO;
+
+      const jobsDone = await Promise.all(jobDoneUuids.map(JobDoneDAL.getJobDone));
+
+      const plansForNextPeriod = await Promise.all(planForNextPeriodUuids.map(PlanForNextPeriodDAL.getPlanForNextPeriod));
+
+      const comments = await Promise.all(commentUuids.map(CommentDAL.getComment));
+
+      const problemsForCurrentPeriod = await Promise.all(problemForCurrentPeriodUuids.map(CurrentProblemDAL.getCurrentProblem));
```

Comment:
looks like this chain includes 7 requests!!! Is it ok?

Reply:
Yes it need to get business model, not DTO. Added comment about finding methods for optimizations

Reply:
7 * 400 ms = 3 sec, it's impossible to wait 3 sec. Parallelize some of them